modules/encoders/x64/xor_context_win7_domainname.rb

```diff
@@ -0,0 +1,173 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Encoder::Xor
+
+  def initialize
+    super(
+      'Name'             => 'Domainname-based Context Keyed Payload Encoder',
+      'Description'      => 'Context-Keyed Payload Encoder based on Domainname and x64 XOR encoder.  Uses an 8 byte key.  So far only tested on Windows 7 x64.',
+      'Author'           => [ 'sf' 'oso', 'jwe' ],
+      'Arch'             => ARCH_X64,
+      'License'          => MSF_LICENSE,
+      'Platform'         => 'windows',
+      'Decoder'          =>
+        {
+          'KeySize'      => 8,
+          'KeyPack'      => 'Q',
+          'BlockSize'    => 8
+        }
+      )
+
+    register_options([ OptString.new('C_DOMAINNAME',[ true, "Context Domainname.", "domainname"])])
+  end
+
+  def obtain_key(buf, badchars, state)
```

Comment:
Am I missing something?  `badchars` and `buf` do not seem to be used in this method?

Reply:
Regarding the buf and badchars, they are leftovers from the original
encoder that I copied to create this one.  Indeed they are not needed.